### ISSUE TITLE
rtpengine iptables duplicates

### ISSUE THREAD

**Issue by powerpbx** [irontec/ivozprovider#843]
When rules are saved/persistent you can end up with one or more duplicate rtpengine rules.  One way around it would be to run a bash script during  [systemd](https://github.com/irontec/rtpengine/blob/bleeding/debian/ivozprovider-rtpengine.rtpengine.service) ExecStartPost/ExecStopPost instead of adding the rules directly.

Another way would be to just delete any existing rule first.  Not sure if it is fullproof but it would be an improvement.

```
[Unit]
Description=RTPEngine Mediaproxy
After=network.target netfilter-persistent.service firewalld.service 

[Service]
Type=forking
EnvironmentFile=/etc/default/rtpengine
PIDFile=/var/run/rtpengine.pid
ExecStart=/usr/sbin/rtpengine --table=0 --interface=${AUDIO_SOCK} --listen-ng=${CONTROL_SOCK} --listen-cli=127.0.0.1:9900 --xmlrpc-format=1 --b2b-url=http://users.ivozprovider.local:8000/RPC2 --tos=184 --pidfile=/var/run/rtpengine.pid --no-fallback --sip-source --recording-dir=${RECORDINGS_SPOOL} --recording-method=proc $EXTRA_OPTS
ExecStartPost=-/sbin/iptables -D INPUT -p udp -j RTPENGINE --id 0
ExecStartPost=/sbin/iptables -I INPUT -p udp -j RTPENGINE --id 0
ExecStopPost=-/sbin/iptables -D INPUT -p udp -j RTPENGINE --id 0
ExecStopPost=/bin/bash -c "/bin/echo del 0 > /proc/rtpengine/control"
Restart=always

[Install]
WantedBy=multi-user.target
```

**Comment by Kaian** [comment]
Hi @powerpbx 

I'm trying to reproduce this problem but without success so far.

According to logs, rtpengine should stops before iptables-persistent (based on its dependency on network-pre.target). I have tested in my development environment and I can confirm that. This means that, even if you're saving rules on shutdown, the rtpengine rules should not be there (because the service have already stopped).

The only possibility to end up with duplicate iptables rules is that they've been saved using `netfilter-persistent save` manually or during the package installation. This will produce incorrect configuration, so you should fix your saved rules to remove other services iptables configuration. Check saved rules with:

```
# grep RTPENGINE /etc/iptables/rules.*
```

If you need to update any service configuration file (rtpengine or any other) for any additional software compatibility, you can also use service configuration directory to change a unit-file behaviour.

For example, consider adding a file in /etc/systemd/system/rtpengine.service.d/my-custom-serviceconf
```
[Unit]
Before=my-custom.service
```
This file will be merged with rtpengine unit file after a daemon-reload

Regards and thanks for reporting!!!

**Comment by powerpbx** [comment]
Yes, we save the rules manually.  That is normally how people do it as far as I know.

**Comment by cruzccl** [comment]
Parcially merged in irontec/rtpengine@120f31efd1e61a880cec5d764b9768c64d735e82
